CNV/somatic/crsp_and_wgs/docker/plot_bp_concordance_pcawg_pilot.py

```diff
@@ -0,0 +1,275 @@
+import math
+
+import matplotlib
+# Force matplotlib to not use any Xwindows backend.
+from pandas.core.frame import DataFrame
+from pandas.core.series import Series
+
+matplotlib.use('Agg')
+
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+import os
+import matplotlib.colors
+import matplotlib.patches as patches
+import matplotlib.pyplot as plt
+import numpy
+import pandas
+from sklearn.metrics import mean_absolute_error as mae
+
+CR_GUESS_COLUMN = "LOG2_COPY_RATIO_POSTERIOR_50"
+PURITY_PLOIDY_FILENAME = "consensus.20170217.purity.ploidy.txt"
+UUID_BARCODE_FILE = "UUID-BARCODE.tsv"
+GT_CN_COLUMN_NAME = "final_total_cn"
+GT_CR_COLUMN_NAME = GT_CN_COLUMN_NAME.replace("cn", "cr")
+MIN_SEGMENT_LENGTH_BP = 0
+MIN_STARS = 2.5
+PREVALENCE_MIN_FOR_BLACKLIST = 0.1
+FRACMATCH_MIN = 0.95
+
+"""
+# Purity/ploidy files uses the same sample names as PCAWG, not TCGA, so the samplename column will match the ground truth segment files.
+
+purity_ploidy_filename = "consensus.20170217.purity.ploidy.txt"
+
+purity_ploidy_df = pandas.read_csv(purity_ploidy_filename, sep="\t")
+
+#sample_of_interest = "5a7f3c88-ef02-4e97-8009-5622245b6a09"
+#input_tsv = "G25783.TCGA-55-6986-01A-11D-1945-08.2.bam.tsv"
+sample_of_interest = "d432e99a-67fb-4609-b90f-99438eee7cae"
+# input_tsv = "G39019.TCGA-DA-A1I2-06A-21D-A19A-08.1.bam.tsv"
+input_tsv = "G39019.TCGA-DA-A1I2.tagged.seg.final.tsv.final.tsv"
+
+purity = float(purity_ploidy_df[purity_ploidy_df.samplename == sample_of_interest].purity)
+ploidy = float(purity_ploidy_df[purity_ploidy_df.samplename == sample_of_interest].ploidy)
+print("sample: " + sample_of_interest)
+print("purity = " + str(purity))
+print("ploidy = " + str(ploidy))
+"""
+
+def calculate_mad_with_weights(comparison_df):
```

Comment:
Done for all python scripts.

Reply:
Oh sorry I wasn't specific - I meant the type hints as described here: https://docs.python.org/3/library/typing.html

I don't think it's necessary especially since you added docs to every function!